server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -0,0 +1,99 @@
+package cached_mysql
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	redigo "github.com/gomodule/redigo/redis"
+	"github.com/pkg/errors"
+)
+
+type cachedMysql struct {
+	fleet.Datastore
+
+	redisPool fleet.RedisPool
+}
+
+const (
+	CacheKeyAppConfig = "AppConfig"
+)
+
+func New(ds fleet.Datastore, redisPool fleet.RedisPool) fleet.Datastore {
+	return &cachedMysql{
+		Datastore: ds,
+		redisPool: redisPool,
+	}
+}
+
+func (ds *cachedMysql) storeInRedis(key string, v interface{}) error {
+	conn := ds.redisPool.ConfigureDoer(ds.redisPool.Get())
+	defer conn.Close()
+
+	b, err := json.Marshal(v)
+	if err != nil {
+		return errors.Wrap(err, "marshaling object to cache in redis")
+	}
+
+	if _, err := conn.Do("SET", key, b); err != nil {
```

Comment:
Should this have an expiration just to be safe, i.e. to prevent both sources to differ over time for whatever reason (one or the other is down for a bit, restoring from a backup, etc.)? Could be quite big like a day, or an hour, whatever makes sense.

Reply:
I can't think of a scenario where they would differ, but I agree it's better to be safe. I'll set it to 1 day for now.